src/devices/HuskyLens/HuskyLens.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Iot.Device.HuskyLens
+{
+    /// <summary>
+    /// todo
+    /// </summary>
+    public class HuskyLens
+    {
+        private readonly IBinaryConnection _connection;
+
+        /// <summary>
+        /// Algorithm currently used.
+        /// </summary>
+        public Algorithm Algorithm { get; private set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="connection">guess</param>
+        public HuskyLens(IBinaryConnection connection)
+        {
+            _connection = connection;
+            Algorithm = Algorithm.Undefined;
+        }
+
+        /// <summary>
+        /// Ping the thing
+        /// </summary>
+        /// <returns>true for success, guess the rest</returns>
+        public bool Ping()
+        {
+            _connection.Write(new byte[] { 0x55, 0xAA, 0x11, 0x00, 0x2C, 0x3C });
+            WaitForOK();
+            return true;
+        }
+
+        /// <summary>
+        /// Sets the currently active algorithm
+        /// </summary>
+        /// <param name="algorithm">the algorithm</param>
+        public void SetAlgorithm(Algorithm algorithm)
```

Comment:
Why are you not using this Set with the Algorithm property? You'll then just get a property Algorithm with a set and a get. A bit simpler from my point of view.